src/commands/force/user/permsetlicense/assign.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as os from 'os';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { Aliases, Messages, SfdxError, User, UserFields } from '@salesforce/core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-user', 'permsetlicense.assign');
+
+type SuccessMsg = {
+  name: string;
+  value: string;
+};
+
+type FailureMsg = {
+  name: string;
+  message: string;
+};
+
+export type PSLResult = {
+  successes: SuccessMsg[];
+  failures: FailureMsg[];
+};
+
+interface PermissionSetLicense {
+  Id: string;
+}
+
+export class UserPermsetLicenseAssignCommand extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessage('examples').split(os.EOL);
+  public static readonly requiresUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    name: flags.string({
+      char: 'n',
+      description: messages.getMessage('flags.name'),
+      required: true,
+    }),
+    onbehalfof: flags.array({
+      char: 'o',
+      description: messages.getMessage('flags.onBehalfOf'),
+    }),
+  };
+  private readonly successes: SuccessMsg[] = [];
+  private readonly failures: FailureMsg[] = [];
+  private pslId: string;
+
+  public async run(): Promise<PSLResult> {
+    const usernames = (this.flags.onbehalfof as string[]) ?? [this.org.getUsername()];
+    this.logger.debug(`will assign permset to users: ${usernames.join(', ')}`);
+    const pslName = this.flags.name as string;
+
+    const conn = this.org.getConnection();
+    try {
+      this.pslId = (
+        await conn.singleRecordQuery<PermissionSetLicense>(
+          `select Id from PermissionSetLicense where DeveloperName = '${pslName}' or MasterLabel = '${pslName}'`
```

Comment:
This is some platform knowledge that I don't have. But I'm assuming that this query _should_ always return a singular result correct? That `DeveloperName` and `MasterLabel` are mutually exclusive

Reply:
In the org, what would lead to this returning multiple results?

Reply:
typically, when you name a PS in the UI, you'll say "my ps name" and it creates `my_ps_name` (api name based on the label).

Now that you mention it, someone could find a way to screw that up by making one DevName match the Label of a _different_ permission set, but that seemed a decent trade for the convenience of being able to user either to find what you wnat.

Reply:
Multiple results would be a bit weird.  No results will be more common.